include/openthread/border_routing.h

```diff
@@ -498,6 +498,32 @@ void otBorderRoutingDhcp6PdSetEnabled(otInstance *aInstance, bool aEnabled);
  */
 otBorderRoutingDhcp6PdState otBorderRoutingDhcp6PdGetState(otInstance *aInstance);
 
+/**
+ * When the state of a DHCPv6 Prefix Delegation (PD) on the Thread interface changes, this callback notifies processes
+ * in the OS of this changed state.
+ *
+ * @param[in] aRequest  Whether to request RA.
```

Comment:
What does it mean to "request RA"? Is the handler of this callback intended to send a Router Solicitation message on the infrastructure link?

Reply:
@sherysheng I recommend:

- Let's change `bool aRequest` to `otBorderRoutingDhcp6PdState aState` in the callback for better type clarity.
- This callback's purpose will be to signal changes to the DHCPv6 Prefix Delegation (PD) state and provides the new state as input.
- Let's omit platform-specific action notes. This keeps the documentation focused on the functionality provided by the OT stack and gives platform developers flexibility in their implementations.

Reply:
I don't think I need to expose the state here as we already have a state API which could be called externally.  The purpose was notifying  the platform when the state changed, and trigger the request/withdraw RA behaviour, use aState here seems creating more trouble to users?

Reply:
Hi Abtin,
Could you explain more about removing the aRequest?  Do you want users to call both the callback API and the state API to decide if they want to withdraw/request RA? I prefer to just use this callback to tell users what they should do instead of using two APIs

Reply:
My suggestion is to help address Jonathan's comment regarding `aRequest` and the "Whether to request RA" documentation.

Given that PR #10000 introduces an alternative way to specify the DHCP6 prefix (no necessarily using RAs), let's keep the callback's purpose centered on signaling DHCP6 state changes. We should avoid platform-specific instructions (like "Whether to request RA"), allowing platform developers flexibility in their implementations.

> Do you want users to call both the callback API and the state API to decide if they want to withdraw/request RA? I prefer to just use this callback to tell users what they should do instead of using two APIs

The callback is invoked by the OT stack, not directly by users. Users/platforms are responsible for implementing the callback. As I suggested, including `otBorderRoutingDhcp6PdState aState` as input in the callback eliminates the need for the implementation to query the state using a separate API.

Reply:
Thanks ! Updated